app/controlplane/internal/biz/referrer.go

```diff
@@ -34,6 +35,47 @@ import (
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
+type ReferrerUseCase struct {
+	repo           ReferrerRepo
+	membershipRepo MembershipRepo
+	workflowRepo   WorkflowRepo
+	logger         *log.Helper
+	indexConfig    *conf.ReferrerSharedIndex
+}
+
+func NewReferrerUseCase(repo ReferrerRepo, wfRepo WorkflowRepo, mRepo MembershipRepo, indexCfg *conf.ReferrerSharedIndex, l log.Logger) (*ReferrerUseCase, error) {
```

Comment:
[very nit]

Maybe when the decl is this long, it could be done vertical?
```
func NewReferrerUseCase(
        repo ReferrerRepo,
        wfRepo WorkflowRepo,
        mRepo MembershipRepo,
        indexCfg *conf.ReferrerSharedIndex,
        l log.Logger,
) (
        *ReferrerUseCase,
        error,
) {
...
}
```

Not a big deal thogh.